File: src/types/tables.ts

export interface Column {
  accessor: string;
  header: string;
  type: 'string' | 'number' | 'boolean' | 'date';
}

export interface Table {
  id: string;
  name: string;
  type: 'uploaded' | 'merged';
  createdAt: Date;
  lastAccessed?: Date;
  columns: Column[];
  data: Record<string, any>[];
  parentTables?: string[]; // For merged tables
  rowCount: number;
}

export interface ChatSession {
  id: string;
  tableId: string;
  name: string;
  createdAt: Date;
  messages: {
    id: string;
    role: 'user' | 'system';
    content: string;
    timestamp: Date;
  }[];
}

File: backend/app/models/tables.py

from typing import List, Dict, Any, Optional, Literal
from pydantic import BaseModel, Field
from datetime import datetime

class Column(BaseModel):
    accessor: str
    header: str
    type: Literal["string", "number", "boolean", "date"]

class Table(BaseModel):
    id: str
    name: str
    type: Literal["uploaded", "merged"]
    createdAt: datetime
    lastAccessed: Optional[datetime] = None
    columns: List[Column]
    data: List[Dict[str, Any]]
    parentTables: Optional[List[str]] = None
    rowCount: int
